UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Changeling/ChangelingMain.cs

```diff
@@ -0,0 +1,626 @@
+using Mirror;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using Systems.Character;
+using UnityEngine;
+using Util;
+
+namespace Changeling
+{
+	[DisallowMultipleComponent]
+	public class ChangelingMain : NetworkBehaviour
+	{
+		[Header("Changeling main")]
+		[SyncVar(hook = nameof(SyncChemCount))]
+		private float chem = 25;
+		public float Chem => chem;
+		[SyncVar(hook = nameof(SyncMindID))] private uint changelingMindID = 0;
+		[SyncVar] private float chemMax = 75;
+		[SyncVar] private float chemAddPerTime = CHEM_ADD_TIME;
+		[SyncVar] private float chemAddTime = CHEM_ADD_PER_TIME_BASE;
+		public int ExtractedDna => changelingDnas.Count;
+
+		[SyncVar(hook = nameof(SyncResetCounts))] private int resetCount = 0;
+		[SyncVar(hook = nameof(SyncMaxResetCounts))] private int resetCountMax = 1;
+
+		[SyncVar] private int absorbCount = 0;
+		public int AbsorbCount => absorbCount;
+		public int ResetsLeft => resetCountMax - resetCount;
+
+		private List<ChangelingDna> changelingDnas = new List<ChangelingDna>();
+		private List<ChangelingMemories> changelingMemories = new List<ChangelingMemories>();
+		public List<ChangelingMemories> ChangelingMemories => changelingMemories;
+
+		public bool isFakingDeath = false;
```

Comment:
This an other public fields you have here are a good example of something you want other classes outside to get but not really interested in setting them in editor o allowing other classes to change it.

Use properties for these cases, you can make a public get private set property like this:

public bool IsFakingDeath {get; private set;}

This will keep setting the internal state of this object to itself. Think of objects as closed organisms, you can't change their internals but you can get information from them and act on them which as a consequence can change their internals but they did it, not me.

Reply:
Ok, I`m just create new method for setting that bool